### DISCUSSION TITLE
Why do we need to use cy.wait(...) on a backendless environment ?

### DISCUSSION THREAD

**Discussion by m719** [cypress-io/cypress#16829] [Questions and Help]
Hello folks,

I have a completely backendless test environment with .json fixtures representing my data.
In the `before()` hook of my tests, I mock all the calls that the test will do using `cy.route(...)` (cypress 5.x).
Since all my XHR calls are stubbed in the beggining of my tests, I was wondering why I do need to use `cy.wait(<insert alias of cy.route here>)` to reduce flakiness.
I would expect `cy.wait(<insert alias of cy.route here>) ` to be only needed for testing involving an actual backend.

I'm afraid I don't fully understand the way stubbed routes work, hence my doubt on `cy.wait(...)` usage.
Could you please help me understand what is wrong in my reasoning ?

Thank you

**Comment by jennifer-shehane** [comment]
`cy.route()` (or the new `cy.intercept()`) is essentially setting up a listener for a request. It doesn't mean that the request ever actually happened. 

Using `cy.wait()` ensures that the request you were listening for has resolved. That way you can move on to your next Cypress command knowing the request has resolved, so the data for your app or proper state, etc is populated.

I would suggest transitioning to `cy.intercept()` since `cy.route()` is now deprecated and read our [Network Requests](https://docs.cypress.io/guides/guides/network-requests) guide.
